onnxruntime/core/providers/cpu/ml/treeregressor.cc

```diff
@@ -55,8 +75,20 @@ common::Status TreeEnsembleRegressor<T>::Compute(OpKernelContext* context) const
   }
   int64_t N = X->Shape().NumDimensions() == 1 ? 1 : X->Shape()[0];
   Tensor* Y = context->Output(0, {N, tree_ensemble_.n_targets_or_classes_});
-
-  tree_ensemble_.compute(context, X, Y, NULL);
+  ORT_IF_CONSTEXPR(std::is_same<TH, TO>::value) {
+    TH* y_data = Y->template MutableData<TH>();
+    tree_ensemble_.compute(context, X, y_data, NULL);
+  } else {
+    int64_t size = Y->Shape().Size();
+    std::unique_ptr<TH> y_data_th(new TH[size]);
+    TH* tree_output = y_data_th.get();
+    tree_ensemble_.compute(context, X, tree_output, NULL);
+    TO* y_data = Y->template MutableData<TO>();
+    for (int64_t i = 0; i < size; ++i) {
+      y_data[i] = static_cast<TO>(tree_output[i]);
+    }
+    y_data_th.release();
```

Comment:
>y_data_th.release(); [](http://example.com/codeflow?start=4&length=20)

This would leak the buffer, would it not?

Reply:
fixed